src/mixins/ready-mixin.ts

```diff
@@ -0,0 +1,64 @@
+import { BooleanIndicator } from 'state-switch'
+
+import {
+  log,
+} from '../config.js'
+
+import type { PuppetSkeleton } from '../puppet/puppet-skeleton.js'
+
+const readyMixin = <MixinBase extends typeof PuppetSkeleton>(mixinBase: MixinBase) => {
+
+  abstract class ReadyMixin extends mixinBase {
+
+    readyIndicator: BooleanIndicator
```

Comment:
Here I change a boolean to the `BooleanIndicator`.

The beauty of this `BooleanIndicator` is that, you can use the below code to wait the desired state:

```ts
await puppet.readyIndicator.ready(true)
```

If the `readyIndicator` is not in the `true` status, then it will wait until it is set to `true` before the returned promise resolved.

Reply:
that is a great feature.